### DISCUSSION TITLE
Better way to use user for testing

### DISCUSSION THREAD

**Discussion by FreePhoenix888** [deep-foundation/Discussions#20] [Q&A] [answered]
# 1 variant
Install `@deep-foundation/payments-tinkoff-c2b` and `@deep-foundation/payments-tinkoff-c2b-test`. Create a `User` and call it `UserForTestingPaymentsTinkoffC2B`, give it permission of the current user. Test in the testing user

# 2 variant

Create a `User` and call it `UserForTestingPaymentsTinkoffC2B`, give it permission of the current user. Install `@deep-foundation/payments-tinkoff-c2b` and `@deep-foundation/payments-tinkoff-c2b-test`. Test in the testing user.

# 3 variant 
Install `@deep-foundation/payments-tinkoff-c2b`. Create a `User` and call it `UserForTestingPaymentsTinkoffC2B`, give it permission of the current user. Install `@deep-foundation/payments-tinkoff-c2b-test` in the testing user. Test in the testing user

**Comment by konard** [comment]
Why do you need to use User?
What these packages do?

**Reply by FreePhoenix888** [reply]
@Konard ,
> What these packages do?

`@deep-foundation/payments-tinkoff-c2b` provides functionality to do payments in tinkoff.
`@deep-foundation/payments-tinkoff-c2b-test` tests that package.

> Why do you need to use User?
I would like to create a separate user where I will add all the data I need for tests (route, port, ...) and then when the tests pass either successfully or non-successfully I will delete that user and everything contained in it

Right now I do not do that and after I use tests I have a lot of links in the user's space that I do not need. By the way there are a lot of HandleRoutes (or how are they called, I do not remember) for the same port/route/handler-code. How does it impact perfomance?

**Reply by konard** [reply]
@FreePhoenix888 if the logic of `HandleRoute` is different, it should be separate. We should not think about performance until we get a real and measurable problem with performance. Deep now is not about performance right now, but about features and reliability. LinksPlatform tries to optimize everything because it is known in advance that on basic operations with links will be high load.

**Reply by konard** [reply, marked as answer]
@FreePhoenix888 
For using the user, I suggest these steps:
1. Install all packages.
2. Create a User, create a Join link from that user to Admin.
3. Start tests using jwt of that User
4. After tests are complete, delete User and its join links with everything that was created by tests.

This is the simplest thing we can do at the moment without setting up specific permission rules.
